tools/lib/node.coffee

```diff
@@ -0,0 +1,62 @@
+semver = require('semver')
+utils = require('./utils')
+
+RESIN_URL_PREFIX = 'http://resin-packages.s3.amazonaws.com/node/'
+NODEJS_URL_PREFIX = 'http://nodejs.org/dist/'
+
+UNSUPPORTED = 'UNSUPPORTED'
+
+ALPINE_ARCH =
+	'amd64': 'alpine-amd64'
+	'i386': 'alpine-i386'
+	'armv6hf': 'alpine-armhf'
+	'armv7hf': 'alpine-armhf'
```

Comment:
It would fit nicely if we had a dict for `arch`. You would just define a field, like `alpine_arch`, and later access it like `$arch.alpine_arch`.
But we don't have such dict.
We could put it in the device types dict, but then we'll have a bunch of duplicates for the devices that share the same arch.

We can improve `doxx` to have a new feature with dicts `FK`s, where the device type dict will reference the `arch` dict, which won't be used directly for templates expansion.
Like the `arch` dict would keep the `arch` details, like alpine mapping, and maybe some debian fixups we have below. And the device type dict will reference it like this: `$arch: "$arch.amd64"`. This reference will be expanded by `metalsmith-dynamic` and replaced with an actual object from the `$arch` dict.